libcaf_core/caf/detail/ring_buffer.test.cpp

```diff
@@ -0,0 +1,115 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/detail/ring_buffer.hpp"
+
+#include "caf/test/caf_test_main.hpp"
+#include "caf/test/test.hpp"
+
+using namespace caf;
+
+using detail::ring_buffer;
+
+TEST("push_back adds element") {
+  ring_buffer<int> buf(3);
+  info("full capacity of ring buffer");
+  for (int i = 1; i <= 3; ++i) {
+    buf.push_back(i);
+  }
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 1);
+  info("additional element after full capacity");
+  buf.push_back(4);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 2);
+  buf.push_back(5);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 3);
+  buf.push_back(6);
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 4);
+}
+
+TEST("pop_front removes element") {
+  ring_buffer<int> buf(3);
+  info("full capacity of ring buffer");
+  for (int i = 1; i <= 3; ++i) {
+    buf.push_back(i);
+  }
+  check_eq(buf.full(), true);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 1);
+  info("remove element from buffer");
+  buf.pop_front();
+  check_eq(buf.full(), false);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 2);
+  buf.pop_front();
+  check_eq(buf.full(), false);
+  check_eq(buf.empty(), false);
+  check_eq(buf.front(), 3);
+  buf.pop_front();
+  check_eq(buf.empty(), true);
+}
+
+TEST("circular buffer functions properly") {
```

Comment:
Can you please give names to tests that gives a better idea on what is actually being tested? Without a definition for "properly", this test name basically says nothing.